From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Behan Webster <behanw@converseincode.com>
Date: Tue, 3 Sep 2013 22:27:25 -0400
Subject: [PATCH] {LLVMLinux} kbuild: LLVMLinux: Clang doesn't support the gcc
 flags set by CONFIG_ARCH_HWEIGHT_CFLAGS

Ignore the flags set in Kconfig for hweight if the compiler is clang.

This is also reported as LLVM PR9457

Author: Behan Webster <behanw@converseincode.com>
Signed-off-by: Behan Webster <behanw@converseincode.com>
---
 lib/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Makefile b/lib/Makefile
index a7c26a41a738..45f078784d6d 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -59,7 +59,9 @@ obj-$(CONFIG_CHECK_SIGNATURE) += check_signature.o
 obj-$(CONFIG_DEBUG_LOCKING_API_SELFTESTS) += locking-selftest.o
 
 GCOV_PROFILE_hweight.o := n
+ifneq ($(cc-name),clang)
 CFLAGS_hweight.o = $(subst $(quote),,$(CONFIG_ARCH_HWEIGHT_CFLAGS))
+endif
 obj-$(CONFIG_GENERIC_HWEIGHT) += hweight.o
 
 obj-$(CONFIG_BTREE) += btree.o
-- 
